lib/manager/npm/extract/locked-versions.ts

```diff
@@ -39,6 +39,10 @@ export async function getLockedVersions(
         logger.trace('Retrieving/parsing ' + npmLock);
         lockFileCache[npmLock] = await getNpmLock(npmLock);
       }
+      const { lockfileVersion } = lockFileCache[npmLock];
+      if (lockfileVersion >= 2) {
+        packageFile.constraints.npm = '>= 7.0.0';
```

Comment:
What's the end result if there is an npm constraint in `engines`?

Reply:
It'd override the `engines` field, as this is in the post-extract step. This is the same behavior with Yarn.

Reply:
I think we maybe made a mistake in Yarn then. If there's something in `engines` then we should honor that unless overridden in constraints *config*.

Reply:
Updated to not override when `constraints` is already set.

Reply:
Correction: the constraint works differently with Yarn 2, it doesn't install the specified version but is only used in feature detection. So overwriting constraints doesn't matter.